packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -159,6 +159,12 @@ export const $lib = createTypeSpecLibrary({
         default: `Cannot pass an empty value to the @clientName decorator`,
       },
     },
+    "override-method-parameters-mismatch": {
+      severity: "error",
+      messages: {
+        default: paramMessage`Method "${"methodName"}" is not directly referencing the same parameters as in the original operation. The original method has parameters "${"originalParameters"}", while the override method has parameters "${"overrideParameters"}".`,
```

Comment:
Nice! Thanks for updating this message as it makes troubleshooting a lot easier now.